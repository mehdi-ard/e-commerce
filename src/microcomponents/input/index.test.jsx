import React from "react";
import renderer from "react-test-renderer";
import Input from "./";

class TestInput extends React.Component {
  state = {
    name: ""
  };

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };
  render() {
    const { name } = this.state;
    return (
      <Input
        id="outlined-password-input"
        label="text"
        type="text"
        autoComplete="current-password"
        margin="normal"
        variant="outlined"
        onChange={this.handleChange("name")}
        value={name}
      />
    );
  }
}

it("renders default", () => {
  expect(renderer.create(<TestInput />).toJSON()).toMatchSnapshot();
});
