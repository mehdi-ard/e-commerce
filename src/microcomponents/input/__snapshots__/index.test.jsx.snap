// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root Input-root-1  MuiFormControl-marginNormal"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
    data-shrink={false}
    htmlFor="outlined-password-input"
  >
    text
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-58 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-59"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
    <input
      aria-invalid={false}
      autoComplete="current-password"
      className="MuiInputBase-input MuiOutlinedInput-input"
      disabled={false}
      id="outlined-password-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
  </div>
</div>
`;
